biz.aQute.repository/src/aQute/bnd/repository/p2/provider/P2Indexer.java

```diff
@@ -93,14 +96,34 @@ File get(String bsn, Version version, Map<String,String> properties, DownloadLis
 		URI url = contentCapability.url();
 
 		final File source = client.getCacheFileFor(url);
+		final Path sourcePath = source.toPath();
 		final File link = new File(location, bsn + "-" + version + ".jar");
-		if (link.isFile())
-			Files.delete(link.toPath());
-		Files.createLink(link.toPath(), source.toPath());
+		final Path linkPath = link.toPath();
 
 		Promise<File> go = client.build().useCache(MAX_STALE).async(url.toURL()).map(new Function<File,File>() {
 			@Override
 			public File apply(File t) {
+				// links can fail if they span disparate Windows drives
+				// so we need to handle them slightly differently
+				try {
+					if (link.isFile() && Files.isSymbolicLink(linkPath))
```

Comment:
Yeah i don't know what happens, if you try to create a symlink to a file that already exists.  It might not through a FileSystemException but another type of IOException.